Form-Builder/src/app/app.component.html

```diff
@@ -0,0 +1,71 @@
+<form class="form" [formGroup]="profileForm" (ngSubmit)="onSubmit()">
+
+  <div class="field">
+    <label class="label">username</label>
+    <input type="text" class="input" formControlName="username" 
+    [ngClass]= "{'invalid': profileForm.get('username')?.invalid && (profileForm.get('username')?.touched || profileForm.get('username')?.dirty)}"/>
+    <div *ngIf="profileForm.get('username')?.hasError('required') && (profileForm.get('username')?.touched || profileForm.get('username')?.dirty)"
+    [ngClass]="{'error': profileForm.get('username')?.hasError('required')}">
+      Username must be at least 3 characters.
+    </div>
+  </div>
+
+  <div class="field">
+    <label class="label">email</label>
+    <input type="text" class="input" formControlName="email" 
+    [ngClass]= "{'invalid': profileForm.get('email')?.invalid && (profileForm.get('email')?.touched || profileForm.get('email')?.dirty)}"/>
```

Comment:
Create a function that checks all these conditions and returns only a boolean to simplify reading the html code